gems/smithy-client/lib/smithy-client/output.rb

```diff
@@ -13,6 +13,9 @@ def initialize(options = {})
         @context = options[:context] || HandlerContext.new
         @data = options[:data]
         @error = options[:error]
+        @request = @context.request
+        @response = @context.response
+        @response.on_error { |error| @error = error }
```

Comment:
This feels a bit round about to use the event mechanism to set the error on the output instead of having the http handler set it.

Reply:
This is so we can signal error to populate the error. Otherwise I think the output object has to be constructed ahead of time and passed around the plugin.

Reply:
And if we're not signaling errors, listeners won't fire off. So we have to signal anyway.